src/components/DeviceStatusGauge.js

```diff
@@ -0,0 +1,241 @@
+import { h } from 'preact'
+import _ from 'lodash'
+import Pie from 'paths-js/pie'
+import styled, { withTheme } from 'styled-components'
+import { compose } from 'recompose'
+import ResinDeviceStatus from 'resin-device-status'
+
+import hoc from '../hoc'
+import theme from '../theme'
+
+const { colors, font } = theme
+
+const DeviceStatusGaugeWrapper = styled.div`
+  display: block;
+
+  .device-status-gauge__graph {
+    display: block;
+    width: 100%;
+    max-width: 180px;
+    margin: 0 auto 10px;
+  }
+
+  .device-status-gauge__graph__total-devices-label {
+    font-family: ${font};
+    font-size: 5px;
+    fill: #b8b8b8;
+    text-anchor: middle;
+  }
+
+  .device-status-gauge__graph__total-devices-number {
+    font-family: ${font};
+    font-size: 26px;
+    fill: #5e5e5e;
+    letter-spacing: -1px;
+    text-anchor: middle;
+  }
+
+  .device-status-gauge__statuses {
+    .clearfix;
+    list-style: none;
+    padding: 5px;
+    margin: 0 auto;
+    background: white;
+    border-radius: 5px;
+    max-width: 240px;
+    border: 1px solid #ececec;
+    overflow: hidden;
+  }
+
+  .device-status-gauge__status {
+    display: flex;
+    padding: 3px 0 3px 2px;
+    line-height: 9px;
+    width: 50%;
+    float: left;
+  }
+
+  .device-status-gauge__status--post-provisioning {
+    width: 100%;
+  }
+
+  .device-status-gauge__status__color {
+    display: inline-block;
+    flex: 0 0 auto;
+    width: 10px;
+    height: 10px;
+    margin-right: 4px;
+  }
+
+  .device-status-gauge__status__name {
+    font-size: 11px;
+  }
+  .block-status-idle {
+    background-color: ${colors.statusIdle.main};
+  }
+  .block-status-configuring {
+    background-color: ${colors.statusConfiguring.main};
+  }
+  .block-status-updating {
+    background-color: ${colors.statusUpdating.main};
+  }
+  .block-status-post-provisioning {
+    background-color: ${colors.statusPostProvisioning.main};
+  }
+  .block-status-offline {
+    background-color: ${colors.statusOffline.main};
+  }
+`
+
+const getDefaultPlaceholderState = () => ({
+  count: 0,
+  key: ResinDeviceStatus.status.OFFLINE,
+  name: _.get(
+    ResinDeviceStatus.statuses.find(
+      s => s.key === ResinDeviceStatus.status.OFFLINE
+    ),
+    'name',
+    'Offline'
+  ),
+  value: 1
+})
+
+const pathFillColor = status => {
+  if (status === 'idle') {
+    return colors.statusIdle.main
+  }
+
+  if (status === 'configuring') {
+    return colors.statusConfiguring.main
+  }
+
+  if (status === 'updating') {
+    return colors.statusUpdating.main
+  }
+
+  if (status === 'post-provisioning') {
+    return colors.statusPostProvisioning.main
+  }
+
+  if (status === 'offline') {
+    return colors.statusOffline.main
+  }
+}
+
+const DeviceStatusGauge = ({ devices, ...props }) => {
+  const data = []
+  const allDevices = devices || []
+  const statuses = ResinDeviceStatus.statuses
```

Comment:
I think we are to tightly coupling components to business logic here. I think it'd be better to think of this as vanilla gauge.

You pass in a data type like array of status and renders a graph. The reason I say this is that there are likely cases where you will need to customize the data the component consumes. You may use this gauge on the admin dashboard, where you want all devices and not just a single users devices, IMO it's better to keep the component lib as dumb and agnostic as possible data in -> component out.

Reply:
I think we should probably offer both, a generic gauge and a device gauge that wraps it.